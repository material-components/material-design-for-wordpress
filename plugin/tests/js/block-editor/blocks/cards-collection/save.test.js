/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { cloneDeep } from 'lodash';

/**
 * Internal dependencies
 */
import Save from '../../../../../assets/src/block-editor/blocks/cards-collection/save';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Save { ...props } /> );
};

const baseProps = {
	attributes: {
		style: 'masonry',
		columns: 4,
		align: 'wide',
		gutter: 16,
		cardsProps: [
			{
				contentLayout: 'text-under-media',
				title: 'Title goes here',
				displayTitle: true,
				secondaryText: 'Secondary text',
				displaySecondaryText: true,
				imageSourceUrl: '',
				isImageEditMode: false,
				displayImage: true,
				supportingText: 'Supporting text',
				displaySupportingText: true,
				primaryActionButtonLabel: 'Button text',
				primaryActionButtonUrl: '',
				primaryActionButtonNewTab: false,
				primaryActionButtonNoFollow: false,
				secondaryActionButtonLabel: 'Button text',
				secondaryActionButtonUrl: '',
				secondaryActionButtonNewTab: false,
				secondaryActionButtonNoFollow: false,
				displayActions: true,
				displaySecondaryActionButton: false,
				outlined: false,
				cornerRadius: 4,
			},
		],
		numberOfCards: 1,
	},
	className: 'test',
};

describe( 'blocks: material/cards-collection: Save', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when style props is `grid`', () => {
		const props = cloneDeep( baseProps );
		props.attributes.style = 'grid';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when style props is `list`', () => {
		const props = cloneDeep( baseProps );
		props.attributes.style = 'list';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
