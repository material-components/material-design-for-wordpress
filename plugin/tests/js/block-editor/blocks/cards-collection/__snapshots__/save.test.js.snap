// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: material/cards-collection: Save matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="test alignwide"
      >
        <div
          class="masonry-grid layout-masonry"
        >
          <div
            class="masonry-grid_column"
            style="width: 25%;"
          >
            <div
              class="card-container"
            >
              <div
                class="mdc-card material-design-card"
                style="border-radius: 4px;"
              >
                <div
                  class="mdc-card__primary-action material-design-card__primary-action mdc-ripple-upgraded"
                  tabindex="0"
                >
                  <div
                    class="material-design-card__primary"
                  >
                    <h2
                      class="material-design-card__title mdc-typography mdc-typography--headline6"
                    >
                      Title goes here
                    </h2>
                    <h3
                      class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
                    >
                      Secondary text
                    </h3>
                  </div>
                  <div
                    class="material-design-card__supporting-text material-design-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
                  >
                    Supporting text
                  </div>
                </div>
                <div
                  class="mdc-card__actions"
                >
                  <div
                    class="mdc-card__action-buttons"
                  >
                    <a
                      class="mdc-button mdc-card__action mdc-card__action--button"
                      href="#"
                    >
                      <div
                        class="mdc-button__ripple"
                      />
                      <span
                        class="mdc-button__label"
                      >
                        Button text
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="test alignwide"
    >
      <div
        class="masonry-grid layout-masonry"
      >
        <div
          class="masonry-grid_column"
          style="width: 25%;"
        >
          <div
            class="card-container"
          >
            <div
              class="mdc-card material-design-card"
              style="border-radius: 4px;"
            >
              <div
                class="mdc-card__primary-action material-design-card__primary-action mdc-ripple-upgraded"
                tabindex="0"
              >
                <div
                  class="material-design-card__primary"
                >
                  <h2
                    class="material-design-card__title mdc-typography mdc-typography--headline6"
                  >
                    Title goes here
                  </h2>
                  <h3
                    class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
                  >
                    Secondary text
                  </h3>
                </div>
                <div
                  class="material-design-card__supporting-text material-design-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
                >
                  Supporting text
                </div>
              </div>
              <div
                class="mdc-card__actions"
              >
                <div
                  class="mdc-card__action-buttons"
                >
                  <a
                    class="mdc-button mdc-card__action mdc-card__action--button"
                    href="#"
                  >
                    <div
                      class="mdc-button__ripple"
                    />
                    <span
                      class="mdc-button__label"
                    >
                      Button text
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`blocks: material/cards-collection: Save matches snapshot when style props is \`grid\` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="test alignwide"
      >
        <div
          class="mdc-layout-grid layout-grid"
        >
          <div
            class="mdc-layout-grid__inner"
          >
            <div
              class="card-container mdc-layout-grid__cell--span-3"
            >
              <div
                class="mdc-card material-design-card"
                style="border-radius: 4px;"
              >
                <div
                  class="mdc-card__primary-action material-design-card__primary-action mdc-ripple-upgraded"
                  tabindex="0"
                >
                  <div
                    class="material-design-card__primary"
                  >
                    <h2
                      class="material-design-card__title mdc-typography mdc-typography--headline6"
                    >
                      Title goes here
                    </h2>
                    <h3
                      class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
                    >
                      Secondary text
                    </h3>
                  </div>
                  <div
                    class="material-design-card__supporting-text material-design-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
                  >
                    Supporting text
                  </div>
                </div>
                <div
                  class="mdc-card__actions"
                >
                  <div
                    class="mdc-card__action-buttons"
                  >
                    <a
                      class="mdc-button mdc-card__action mdc-card__action--button"
                      href="#"
                    >
                      <div
                        class="mdc-button__ripple"
                      />
                      <span
                        class="mdc-button__label"
                      >
                        Button text
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="test alignwide"
    >
      <div
        class="mdc-layout-grid layout-grid"
      >
        <div
          class="mdc-layout-grid__inner"
        >
          <div
            class="card-container mdc-layout-grid__cell--span-3"
          >
            <div
              class="mdc-card material-design-card"
              style="border-radius: 4px;"
            >
              <div
                class="mdc-card__primary-action material-design-card__primary-action mdc-ripple-upgraded"
                tabindex="0"
              >
                <div
                  class="material-design-card__primary"
                >
                  <h2
                    class="material-design-card__title mdc-typography mdc-typography--headline6"
                  >
                    Title goes here
                  </h2>
                  <h3
                    class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
                  >
                    Secondary text
                  </h3>
                </div>
                <div
                  class="material-design-card__supporting-text material-design-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
                >
                  Supporting text
                </div>
              </div>
              <div
                class="mdc-card__actions"
              >
                <div
                  class="mdc-card__action-buttons"
                >
                  <a
                    class="mdc-button mdc-card__action mdc-card__action--button"
                    href="#"
                  >
                    <div
                      class="mdc-button__ripple"
                    />
                    <span
                      class="mdc-button__label"
                    >
                      Button text
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`blocks: material/cards-collection: Save matches snapshot when style props is \`list\` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="test alignwide"
      >
        <div
          class="mdc-layout-grid layout-list"
        >
          <div
            class="mdc-layout-grid__inner"
          >
            <div
              class="card-container mdc-layout-grid__cell--span-12"
            >
              <div
                class="mdc-card material-design-card material-design-card__list material-design-basic"
                style="border-radius: 4px;"
              >
                <div
                  class="mdc-card__primary-action material-design-card__primary-action"
                  tabindex="0"
                >
                  <div
                    class="material-design-card__primary"
                  >
                    <h2
                      class="material-design-card__title mdc-typography mdc-typography--headline6"
                    >
                      Title goes here
                    </h2>
                    <h3
                      class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
                    >
                      Secondary text
                    </h3>
                  </div>
                </div>
                <div
                  class="mdc-card__actions"
                >
                  <div
                    class="mdc-card__action-buttons"
                  >
                    <a
                      class="mdc-button mdc-card__action mdc-card__action--button"
                      href="#"
                    >
                      <div
                        class="mdc-button__ripple"
                      />
                      <span
                        class="mdc-button__label"
                      >
                        Button text
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="test alignwide"
    >
      <div
        class="mdc-layout-grid layout-list"
      >
        <div
          class="mdc-layout-grid__inner"
        >
          <div
            class="card-container mdc-layout-grid__cell--span-12"
          >
            <div
              class="mdc-card material-design-card material-design-card__list material-design-basic"
              style="border-radius: 4px;"
            >
              <div
                class="mdc-card__primary-action material-design-card__primary-action"
                tabindex="0"
              >
                <div
                  class="material-design-card__primary"
                >
                  <h2
                    class="material-design-card__title mdc-typography mdc-typography--headline6"
                  >
                    Title goes here
                  </h2>
                  <h3
                    class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
                  >
                    Secondary text
                  </h3>
                </div>
              </div>
              <div
                class="mdc-card__actions"
              >
                <div
                  class="mdc-card__action-buttons"
                >
                  <a
                    class="mdc-button mdc-card__action mdc-card__action--button"
                    href="#"
                  >
                    <div
                      class="mdc-button__ripple"
                    />
                    <span
                      class="mdc-button__label"
                    >
                      Button text
                    </span>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
